File: app/src/main/java/com/nasahapps/materialdesigntoolbox/example/ui/components/StepperHorizonalFragment.java
package com.nasahapps.materialdesigntoolbox.example.ui.components;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import com.nasahapps.materialdesigntoolbox.components.HorizontalStepperLayout;
import com.nasahapps.materialdesigntoolbox.components.Stepper;
import com.nasahapps.materialdesigntoolbox.example.R;
import com.nasahapps.materialdesigntoolbox.example.ui.main.MainActivity;

import butterknife.Bind;
import butterknife.OnClick;

/**
 * Created by Hakeem on 4/17/16.
 */
public class StepperHorizonalFragment extends ComponentFragment {

    @Bind(R.id.layout)
    HorizontalStepperLayout mStepperLayout;

    int mCurrentStep, mMaxStep;

    @Override
    public int getLayoutId() {
        return R.layout.fragment_stepper_horizontal_layout;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mMaxStep = mStepperLayout.getStepperCount();
    }

    @OnClick(R.id.button)
    public void nextStep() {
        if (mCurrentStep >= mMaxStep) {
            // Reset back to step 1
            mCurrentStep = 0;
            for (int i = 0; i < mStepperLayout.getStepperCount(); i++) {
                mStepperLayout.setStepperState(i, Stepper.INACTIVE);
            }

            mStepperLayout.setStepperState(mCurrentStep, Stepper.ACTIVE);
        } else {
            // Increment a step
            mStepperLayout.setStepperState(mCurrentStep, Stepper.COMPLETED);
            mCurrentStep++;
            if (mCurrentStep < mMaxStep) {
                mStepperLayout.setStepperState(mCurrentStep, Stepper.ACTIVE);
            }
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        ((MainActivity) getActivity()).setToolbarTitle("Steppers");
    }
}

File: app/src/main/java/com/nasahapps/mdt/example/ui/components/StepperHorizonalFragment.kt
package com.nasahapps.mdt.example.ui.components

import android.os.Bundle
import android.view.View
import com.nasahapps.mdt.components.Stepper
import com.nasahapps.mdt.example.R
import com.nasahapps.mdt.example.ui.main.MainActivity
import kotlinx.android.synthetic.main.fragment_stepper_horizontal_layout.*

/**
 * Created by Hakeem on 4/17/16.
 */
class StepperHorizonalFragment : ComponentFragment() {

    internal var mCurrentStep: Int = 0
    internal var mMaxStep: Int = 0

    override fun getLayoutId(): Int {
        return R.layout.fragment_stepper_horizontal_layout
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mMaxStep = layout.stepperCount

        button?.setOnClickListener { nextStep() }
    }

    private fun nextStep() {
        if (mCurrentStep >= mMaxStep) {
            // Reset back to step 1
            mCurrentStep = 0
            for (i in 0..layout.stepperCount - 1) {
                layout?.setStepperState(i, Stepper.INACTIVE)
            }

            layout?.setStepperState(mCurrentStep, Stepper.ACTIVE)
        } else {
            // Increment a step
            layout?.setStepperState(mCurrentStep, Stepper.COMPLETED)
            mCurrentStep++
            if (mCurrentStep < mMaxStep) {
                layout?.setStepperState(mCurrentStep, Stepper.ACTIVE)
            }
        }
    }

    override fun onResume() {
        super.onResume()
        (activity as MainActivity).setToolbarTitle("Steppers")
    }
}
